dango/types/src/oracle/price.rs

```diff
@@ -55,3 +82,28 @@ impl TryFrom<PriceFeed> for PrecisionlessPrice {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use grug::NumberConst;
+
+    use super::*;
+
+    #[test]
+    fn value_of_unit_amount_does_not_overflow_with_large_precision() {
+        // $100M per ETH
+        let price = PrecisionedPrice {
+            humanized_price: Udec128::new(100_000_000u128),
+            humanized_ema: Udec128::ONE,
+            timestamp: 0,
+            precision: Defined::new(18),
+        };
+
+        // Value of 100M ETH at $100M = $100000T
+        let value = price
+            .value_of_unit_amount(Uint128::new(100_000_000u128 * 10u128.pow(18)))
+            .unwrap();
+        println!("{value:?}");
```

Comment:
This print statement can be removed?

Reply:
Yes, was just for testing, forgot to remove.